colcon_core/event_reactor.py

```diff
@@ -88,23 +88,36 @@ def _notify_observers(self, event):
 
     def flush(self):
         """Wait until the queue is empty."""
-        while self.is_alive():
+        while self._thread.is_alive():
             if self._queue.empty():
                 return
             time.sleep(0.01)
 
-    def join(self, *args, **kwargs):
+    def stop(self):
         """
-        Join the thread.
+        Stop this event reactor and block until done.
 
         An :class:`EventReactorShutdown` event is added to the queue to notify
-        all observers that the event reactor is about to shut down.
+        all observers that the event reactor is shutting down.
         """
-        logger.debug('joining thread')
         self._queue.put((EventReactorShutdown(), None))
-        super().join(*args, **kwargs)
+        logger.debug('joining thread')
+        self._thread.join()
         logger.debug('joined thread')
 
+    def start(self):
+        """Start the event reactor."""
+        self._thread.start()
+
+    def __enter__(self):
+        """Support for context manager protocol. Start the event reactor."""
```

Comment:
The semantics of `__enter__` should be clear. Please remove the first sentence and just keep the second. The return value should be documented though.

Same below for `__exit__`.